compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -35,20 +34,26 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
         tcx: TyCtxt<'tcx>,
         def_id: LocalDefId,
         typing_env: ty::TypingEnv<'tcx>,
-        get_operand_ty: impl Fn(&hir::Expr<'tcx>) -> Ty<'tcx> + 'a,
+        expr_ty: impl Fn(&hir::Expr<'tcx>) -> Ty<'tcx> + 'a,
+        node_ty: impl Fn(hir::HirId) -> Ty<'tcx> + 'a,
```

Comment:
These are the same function--`hir::Expr` has a `HirId`. No need to have two.

Reply:
The `expr_ty` callback in `fn_ctxt/checks.rs` currently uses the `expr_ty_adjusted`; switching to `node_type` will cause adjustment to be lost.